sdk/src/types/block/slot/commitment.rs

```diff
@@ -87,15 +92,6 @@ impl SlotCommitment {
 
     /// Computes the [`SlotCommitmentId`] of the [`SlotCommitment`].
     pub fn id(&self) -> SlotCommitmentId {
-        let mut bytes = [0u8; SlotCommitmentId::LENGTH];
-        let mut packer = SlicePacker::new(&mut bytes);
-        let content = self.pack_to_vec();
-        let content_hash: [u8; 32] = Blake2b256::digest(content).into();
-
-        // PANIC: packing to an array of bytes can't fail.
-        content_hash.pack(&mut packer).unwrap();
-        self.index.pack(&mut packer).unwrap();
-
-        SlotCommitmentId::from(bytes)
+        SlotCommitmentHash::new(Blake2b256::digest(self.pack_to_vec()).into()).with_slot_index(self.index)
```

Comment:
IMO it's a bit odd that a `with_` method returns another type, maybe there is a more fitting name

Reply:
It's certainly not consistent how `with_` is used, but I can't think of a better option